docs/react/guides/ssr.md

```diff
@@ -288,40 +313,201 @@ ReactDOM.hydrate(
+
+The hooks provided by the `react-query` package need to retrieve a `QueryClient` from their context. Wrap your component tree with `<QueryClientProvider>` and pass it an instance of `QueryClient`.
+
+```tsx
+// app/providers.jsx
+'use client'
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+
+export default function Providers({ children }) {
+  const [queryClient] = React.useState(() => new QueryClient())
+
+  return (
+    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+  )
+}
+```
+
+```tsx
+// app/layout.jsx
+import Providers from './providers'
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <head />
+      <body>
+        <Providers>{children}</Providers>
+      </body>
+    </html>
+  )
+}
+```
+
+### Using `initialData`
+
+Fetch your initial data in a Server Component higher up in the component tree, and pass it to your Client Component as a prop.
+
+```tsx
+// app/page.jsx
+export default async function Home() {
+  const initialData = await getPosts()
+
+  return <Posts posts={initialData} />
+}
+```
+
+```tsx
+// app/posts.jsx
+'use client'
+
+import { useQuery } from '@tanstack/react-query'
+
+export function Posts(props) {
+  const { data } = useQuery({
+    queryKey: ['posts'],
+    queryFn: getPosts,
+    initialData: props.posts,
+  })
+
+  // ...
+}
+```
+
+### Using `<Hydrate>`
+
+Create a request-scoped singleton instance of `QueryClient`. **This ensures that data is not shared between different users and requests, while still only creating the QueryClient once per request.**
+
+```tsx
+// app/getQueryClient.jsx
+import { QueryClient } from '@tanstack/query-core'
+import { cache } from 'react'
+
+const getQueryClient = cache(() => new QueryClient())
+export default getQueryClient
+```
+
+Fetch your data in a Server Component higher up in the component tree than the Client Components that use the prefetched queries. Your prefetched queries will be available to all components deeper down the component tree.
+
+- Retrieve the `QueryClient` singleton instance
+- Prefetch the data using the client's prefetchQuery method and wait for it to complete
+- Use `dehydrate` to obtain the dehydrated state of the prefetched queries from the query cache
+- Wrap the component tree that needs the prefetched queries inside `<Hydrate>`, and provide it with the dehydrated state
+- You can fetch inside multiple Server Components and use `<Hydrate>` in multiple places
+
+> NOTE: TypeScript currently complains of a type error when using async Server Components. As a temporary workaround, use `{/* @ts-expect-error Server Component */}` when calling this component inside another. For more information, see [End-to-End Type Safety](https://beta.nextjs.org/docs/configuring/typescript#end-to-end-type-safety) in the Next.js 13 beta docs.
+
+```tsx
+// app/hydratedPosts.jsx
+import { dehydrate } from '@tanstack/query-core'
```

Comment:
we should really import dehydrate from react-query, as it contains the whole query-core. Users won't have the query-core installed separately